parsers/c.c

```diff
@@ -1777,19 +1777,33 @@ static void skipCppTemplateParameterList (void)
 
 		if (c == '<')
 		{
-			if (roundBracketsLevel == 0)
+			int x = cppGetc ();
+			if(x != '<') 
 			{
-				if (defaultValueExpected == FALSE)
-					++angleBracketsLevel;
+				cppUngetc (x);
+				if (roundBracketsLevel == 0)
+				{
+					if (defaultValueExpected == FALSE)
+						++angleBracketsLevel;
+				}
 			}
+			else if(CollectingSignature)
```

Comment:
> else if (CollectingSignature)

Do we need 'else' here? Only 'if' is not enough?
My understanding is that c must be gather to Signature if CollectingSignature is non-zero.